packages/assets-controllers/src/DeFiPositionsController/fetch-positions.ts

```diff
@@ -0,0 +1,77 @@
+export type DefiPositionResponse = AdapterResponse<{
+  tokens: ProtocolToken[];
+}>;
+
+type ProtocolDetails = {
+  chainId: number;
+  protocolId: string;
+  productId: string;
+  name: string;
+  description: string;
+  iconUrl: string;
+  siteUrl: string;
+  positionType: PositionType;
+};
+
+type AdapterResponse<ProtocolResponse> =
+  | (ProtocolDetails & {
+      chainName: string;
+    } & (
+        | (ProtocolResponse & { success: true })
+        | (AdapterErrorResponse & { success: false })
+      ))
+  | (AdapterErrorResponse & { success: false });
+
+type AdapterErrorResponse = {
+  error: {
+    message: string;
+  };
+};
+
+export type PositionType = 'supply' | 'borrow' | 'stake' | 'reward';
+
+export type ProtocolToken = Balance & {
+  type: 'protocol';
+  tokenId?: string;
+};
+
+export type Underlying = Balance & {
+  type: 'underlying' | 'underlying-claimable';
+  iconUrl: string;
+};
+
+export type Balance = {
+  address: string;
+  name: string;
+  symbol: string;
+  decimals: number;
+  balanceRaw: string;
+  balance: number;
+  price?: number; // TODO: Confirm this is the case
+  tokens?: Underlying[];
+};
+
+// TODO: Update with new API URL
+const DEFI_POSITIONS_API_URL =
```

Comment:
(need to check the docs) but APIs are to be moved into a `services` folder in the controller.

But you've done a great job at isolating these!

Reply:
I cannot see any folder specific instructions for the folder in the controller guidelines.

Happy to drop it in a service folder if that makes things easier.